File: app/src/main/java/com/rk/xededitor/Settings/SettingsData.java
package com.rk.xededitor.Settings;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.res.Configuration;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Iterator;

public class SettingsData {


    public static boolean isDarkMode(Context ctx) {
        return (ctx.getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK)
                == Configuration.UI_MODE_NIGHT_YES;
    }

    public static boolean isOled(Context ctx) {
        return Boolean.parseBoolean(getSetting(ctx, "isOled", "false"));
    }

    public static boolean getBoolean(Context ctx, String key, Boolean Default) {
        return Boolean.parseBoolean(getSetting(ctx, key, Boolean.toString(Default)));
    }

    public static void setBoolean(Context ctx, String key, boolean value) {
        setSetting(ctx, key, Boolean.toString(value));
    }

    public static String getSetting(Context ctx, String key, String Default) {
        SharedPreferences sharedPreferences = ctx.getApplicationContext().getSharedPreferences("Settings", Context.MODE_PRIVATE);
        return sharedPreferences.getString(key, Default);
    }

    public static void setSetting(Context ctx, String key, String value) {
        SharedPreferences sharedPreferences = ctx.getApplicationContext().getSharedPreferences("Settings", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(key, value);
        //editor.apply();
        editor.commit();
    }


}

File: app/src/main/java/com/rk/xededitor/Settings/SettingsData.kt
package com.rk.xededitor.Settings

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import android.content.res.Configuration

@SuppressLint("ApplySharedPref")
object SettingsData {
    private const val PREFNAME = "Settings"

    @JvmStatic
    fun isDarkMode(ctx: Context): Boolean {
        return ((ctx.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK)
                == Configuration.UI_MODE_NIGHT_YES)
    }

    @JvmStatic
    fun isOled(): Boolean {
       return getBoolean(Keys.OLED,false)
    }

    @JvmStatic
    fun getBoolean(key: String?, default: Boolean): Boolean {
        return sharedPreferences!!.getBoolean(key,default)
    }


    @JvmStatic
    fun setBoolean(key: String?, value: Boolean) {
        val editor = sharedPreferences!!.edit()
        editor.putBoolean(key, value)
        editor.commit()
    }

    @JvmStatic
    fun initPref(ctx: Context){
        sharedPreferences = ctx.applicationContext.getSharedPreferences(PREFNAME, Context.MODE_PRIVATE)
    }

    @JvmStatic
    private var sharedPreferences:SharedPreferences? = null

    @JvmStatic
    fun getString(key: String?, default: String): String {
        return sharedPreferences!!.getString(key, default) ?: default
    }

    @JvmStatic
    fun setString(key: String?, value: String?) {
        val editor = sharedPreferences!!.edit()
        editor.putString(key, value)
        editor.commit()
    }
}
